components/public-api-server/pkg/oidc/service.go

```diff
@@ -318,3 +323,25 @@ func (s *Service) CreateSession(ctx context.Context, flowResult *AuthFlowResult,
 	log.WithField("create-session-error", message).Error("Failed to create session (via server)")
 	return nil, message, fmt.Errorf("unexpected status code: %v", res.StatusCode)
 }
+
+func (s *Service) validateRequiredClaims(token *goidc.IDToken) error {
+	if len(token.Audience) < 1 {
+		return fmt.Errorf("audience claim is missing")
+	}
+	var claims struct {
+		Email string `json:"email,omitempty"`
+		Name  string `json:"name,omitempty"`
+		jwt.RegisteredClaims
+	}
```

Comment:
[nit] I know it's not used elsewhere, but I'd hoist this outside of the method to make it a bit more explicit for future readers